File: app/src/main/java/com/wafflestudio/snutt/ui/adapter/IntroAdapter.java
package com.wafflestudio.snutt.ui.adapter;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentStatePagerAdapter;
import android.util.Log;

import com.wafflestudio.snutt.ui.SignInFragment;
import com.wafflestudio.snutt.ui.SignUpFragment;

/**
 * Created by makesource on 2016. 3. 26..
 */
public class IntroAdapter extends FragmentStatePagerAdapter {

    private static final String TAG = "INTRO_ADAPTER";
    private static final int NUM_ITEMS = 2;

    private static SignInFragment signInFragment;
    private static SignUpFragment signUpFragment;

    public IntroAdapter(FragmentManager fm) {
        super(fm);
    }

    /**
     * Return the Fragment associated with a specified position.
     *
     * @param position
     */
    @Override
    public Fragment getItem(int position) {
        switch (position) {
            case 0:
                if (signInFragment == null) signInFragment = SignInFragment.newInstance();
                return signInFragment;
            case 1:
                if (signUpFragment == null) signUpFragment = signUpFragment.newInstance();
                return signUpFragment;
            default:
                Log.e (TAG, "intro fragment position is out of index!");
                return null;
        }
    }

    /**
     * Return the number of views available.
     */
    @Override
    public int getCount() {
        return NUM_ITEMS;
    }

    @Override
    public CharSequence getPageTitle(int position) {
        switch (position) {
            case 0:
                return "로그인";
            case 1:
                return "회원가입";
        }
        return null;
    }
}

File: app/src/main/java/com/wafflestudio/snutt2/adapter/IntroAdapter.kt
package com.wafflestudio.snutt2.adapter

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter
import com.wafflestudio.snutt2.ui.SignInFragment
import com.wafflestudio.snutt2.ui.SignUpFragment

/**
 * Created by makesource on 2016. 3. 26..
 */
class IntroAdapter(fm: FragmentManager?) : FragmentStatePagerAdapter(fm) {
    /**
     * Return the Fragment associated with a specified position.
     *
     * @param position
     */
    override fun getItem(position: Int): Fragment {
        return when (position) {
            0 -> {
                if (signInFragment == null) signInFragment = SignInFragment.newInstance()
                signInFragment!!
            }
            1 -> {
                if (signUpFragment == null) signUpFragment = SignUpFragment.newInstance()
                signUpFragment!!
            }
            else -> {
                throw IllegalStateException("intro fragment position is out of index!")
            }
        }
    }

    /**
     * Return the number of views available.
     */
    override fun getCount(): Int {
        return NUM_ITEMS
    }

    override fun getPageTitle(position: Int): CharSequence? {
        when (position) {
            0 -> return "로그인"
            1 -> return "회원가입"
        }
        return null
    }

    companion object {
        private const val TAG = "INTRO_ADAPTER"
        private const val NUM_ITEMS = 2
        private var signInFragment: SignInFragment? = null
        private var signUpFragment: SignUpFragment? = null
    }
}